### PR TITLE
Rebeka - API

### PR DESCRIPTION
**Pull request by rebkova**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -22,9 +17,80 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// --- ALL THE ROUTES ---
+
+// --- front page ---
+app.get('/', (request /* incoming */, response /* outgoing */) => {
```
Comment (HIPPIEKICK) [inline server.js:23]: These inline comments make it look a little messy 🙈 Maybe keep this in your personal notes instead 😇

File: server.js [hunk 2]
```diff
@@ -22,9 +17,80 @@ const app = express()
...
+// --- return ALL BOOKS & search by TITLE using QUERY SELECTORS ---
+//localhost:8080/books?title=freakonomics
+
+app.get("/books", (request, response) => {
+
+  const titleSearch = request.query.title
+
+  if (titleSearch === undefined) {
+    response.json(booksData)
+  }
```
Comment (HIPPIEKICK) [inline server.js:38]: Hmm, when would the titleSearch be `undefined`?

File: server.js [hunk 3]
```diff
@@ -22,9 +17,80 @@ const app = express()
...
+})
+
+
+// --- getting SPECIFIC book based on ISBN13 ---
+//localhost:8080/id/:isbn13
+
+app.get("/id/:isbn13", (request, response) => {
+
+  const bookId = request.params.isbn13
```
Comment (HIPPIEKICK) [inline server.js:60]: Maybe call it isbn to avoid confusion 😇

File: server.js [hunk 4]
```diff
@@ -22,9 +17,80 @@ const app = express()
...
+      response.status(404).json(`Ups, no keyword called '${titleLowerCase}' in a title.`)
+    }
+  }
+})
+
+
+// --- getting SPECIFIC book based on ISBN13 ---
+//localhost:8080/id/:isbn13
+
+app.get("/id/:isbn13", (request, response) => {
```
Comment (HIPPIEKICK) [inline server.js:58]: Endpoints should be named after what they return and not how they are filtered. This should therefore be named `/books/:isbn13`

File: server.js [hunk 5]
```diff
@@ -22,9 +17,80 @@ const app = express()
...
+    else {
+      return 0
+    }
+  }
+
+  //sorting the numbers 
+  const sortedRatingArray = booksData.sort(compare)
+
+  response.json(sortedRatingArray)
 })
```
Comment (HIPPIEKICK) [inline server.js:94]: Wouldn't it be nice if this was a query under your main endpoint 🤩 
`/books?sort=descending`

**Review by HIPPIEKICK — APPROVED** [review]
Just think about RESTful naming (nouns in plural - name after what the endpoint return) :)

File: server.js (hunk 2 above)
Comment (rebkova) [reply server.js:38]: I think if the title would have no entries i.e. the query would be empty. Like that: `/books?title `or `/books?title=` Correct me if I'm wrong.

File: server.js (hunk 3 above)
Comment (rebkova) [reply server.js:60]: 👍

File: server.js (hunk 4 above)
Comment (rebkova) [reply server.js:58]: 👍

**Pull request closed without merging**